accio-main/src/main/java/io/accio/main/web/DuckDBResource.java

```diff
@@ -0,0 +1,167 @@
+                    return new QueryResultDto(ImmutableList.of(), ImmutableList.of());
+                }
+                throw e;
+            }
+            catch (Exception e) {
+                throw new AccioException(GENERIC_USER_ERROR, e);
+            }
+        }).whenComplete(bindAsyncResponse(asyncResponse));
+    }
+
+    @GET
+    @Path("/settings/init-sql")
+    public void getInitSQL(@Suspended AsyncResponse asyncResponse)
+    {
+        supplyAsync(metadata::getInitSQL)
+                .whenComplete(bindAsyncResponse(asyncResponse));
+    }
+
+    @PUT
+    @Path("/settings/init-sql")
+    public void setInitSQL(
+            String sql,
+            @Suspended AsyncResponse asyncResponse)
+    {
+        runAsync(() -> {
+            String initSQL = metadata.getInitSQL();
+            metadata.setInitSQL(sql);
+            try {
+                metadata.reload();
+            }
+            catch (Exception e) {
+                metadata.setInitSQL(initSQL);
+                throw e;
+            }
+            java.nio.file.Path initSQLPath = metadata.getInitSQLPath();
+            FileUtil.archiveFile(initSQLPath);
+            FileUtil.createFile(initSQLPath, sql);
+        }).whenComplete(bindAsyncResponse(asyncResponse));
+    }
+
+    @PATCH
+    @Path("/settings/init-sql")
+    public void appendToInitSQL(
+            String sql,
+            @Suspended AsyncResponse asyncResponse)
+    {
+        runAsync(() -> {
+            metadata.directDDL(sql);
+            metadata.setInitSQL(metadata.getInitSQL() + "\n" + sql);
+            FileUtil.appendToFile(metadata.getInitSQLPath(), sql);
+        }).whenComplete(bindAsyncResponse(asyncResponse));
+    }
+
+    @GET
+    @Path("/settings/session-sql")
+    public void getSessionSQL(@Suspended AsyncResponse asyncResponse)
+    {
+        supplyAsync(metadata::getSessionSQL)
+                .whenComplete(bindAsyncResponse(asyncResponse));
+    }
+
+    @PUT
+    @Path("/settings/session-sql")
+    public void setSessionSQL(
+            String sql,
+            @Suspended AsyncResponse asyncResponse)
+    {
+        runAsync(() -> {
+            metadata.setSessionSQL(sql);
+            metadata.getClient().closeAndInitPool();
+            java.nio.file.Path sessionSQLPath = metadata.getSessionSQLPath();
+            FileUtil.archiveFile(sessionSQLPath);
+            FileUtil.createFile(sessionSQLPath, sql);
+        }).whenComplete(bindAsyncResponse(asyncResponse));
+    }
+
+    @PATCH
+    @Path("/settings/session-sql")
+    public void appendToSessionSQL(
+            String sql,
+            @Suspended AsyncResponse asyncResponse)
+    {
+        runAsync(() -> {
+            String sessionSQL = metadata.getSessionSQL();
+            metadata.setSessionSQL(sessionSQL + "\n" + sql);
+            try {
+                metadata.getClient().closeAndInitPool();
+            }
+            catch (Exception e) {
+                metadata.setSessionSQL(sessionSQL);
```

Comment:
```suggestion
                metadata.setSessionSQL(sessionSQL);
                metadata.getClient().closeAndInitPool();
```
We should reopen the pool after rollback the session SQL.